_drafts/release-workflows.md

```diff
@@ -0,0 +1,309 @@
+|-------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **push** to `master`                                  | Release on `latest` channel if the commit trigger an allowed release type, report an error otherwise                                                                                                                                                                                                      |
+| **merge** feature branch to `master`                  | Release on `latest` channel if the commit trigger an allowed release type, report an error otherwise                                                                                                                                                                                                      |
+| **push** to `next`                                    | Release on `next` channel                                                                                                                                                                                                                                                                                 |
+| **merge** any branch to `next`                        | Release on `next` channel                                                                                                                                                                                                                                                                                 |
+| **merge** some commits from `next` branch to `master` | Get the last release for commits on branch `master` (including the one that come from `next`), analyze commits, increase the version; If the commit associated with the last release found is also present on `next`, make the version available on the `latest` channel, otherwise do a regular release. |
+
+##### Future branches configuration
+
+```json
+{
+  "release": {
+    "branches": ["master", "next"]
+  }
+}
+```
+
+Internally **semantic-release** will determines the type of commit that can be done on each branch, based on the number of future branches defined:
+- `"branches": ["master"]` => Any type of commit can be pushed to `master`
+- `"branches": ["master", "next"]` => `master` will accept `fix` and `feat` commits; `next` will accept any type of commits.
+- `"branches": ["master", "next", "next-major"]` => `master` will accept `fix`; `next` will accept `fix` and `feat` commits; `next-major` will accept any type of commits.
```

Comment:
I don't think we need that distinction, I'd suggest we just start out with the `["master", "next"]` convention

Reply:
Several project manage different branches for the next breaking and the next feature release.
See https://github.com/sass/sass/blob/stable/CONTRIBUTING.md#branches

Instead of hard-coding things and having to come back to it later, I prefer to implement that properly right away, based on semver. That would also make the code simpler as it would avoid to add extra complexity just to limit capabilities.

Reply:
in that case, I wonder if we should support `master` & `next` as well as `master` & `next-feature` & `next-breaking` but not a mix of the two? That way we avoid that `next` can mean different things based on weather `next-breaking` exists or not.

And I really, really don't want to use minor/major and instead use feature/breaking for all conventions / settings in semantic-release

Reply:
We should support any names via configuration. The question is regarding what we set as default when there is no `branches` configuration.
We can use whatever you prefer as default but I'm not sure if it's really necessary, because those branches would be defined once and never change.

If we don't determine default ones, users would have to create the `next`, `next-feature`, `next-breaking`, or just the `next` and configure them in the `branches` option and never change them again.

Those auto-default would add some complexity in the code and make it more opinionated to just avoid users to do a one time setting. Not sure if it's a good trade-off.

In addition as it's quite a structuring decision to make for the project, it might be better to "force" users to think about it make a deliberate decision.
For example in [condition-travis](https://github.com/semantic-release/condition-travis) we have `next` branch hanging there for some reasons that we never really used. With the auto-default mechanism semantic-release would have considered `master` as the branch for fix and features  only and `next` for breaking changes. Even though we were doing breaking changes on `master` and not using `next`.

Reply:
okay so we would support `master` & `next` by default, everything would require configuration? That sounds like a good compromise

Reply:
ok

Reply:
I fixed my comment above, I left out "else" when I first commented, I want to make sure that does not change the meaning to you

Reply:
yeah I understood it that way

Reply:
> I wonder if we should support master & next as well as master & next-feature & next-breaking but not a mix of the two? That way we avoid that next can mean different things based on weather next-breaking exists or not.

As seen [here](https://github.com/semantic-release/evolution/pull/1/files#r194565577), I actually did find the default of `next` confusing. (Though @pvdlg's explanation was really helpful, and I now know how it's intended to work)

> And I really, really don't want to use minor/major and instead use feature/breaking for all conventions / settings in semantic-release

I agree with that statement as well.